katsdpcontroller/generator.py

```diff
@@ -493,8 +442,9 @@ def _make_fgpu(
 
     for i in range(0, n_engines):
         srcs = stream.sources(i)
-        fgpu = SDPLogicalTask(f'f.{stream.name}.{i}')
+        fgpu = SDPLogicalTask(f'f.{stream.name}.{i}', streams=[stream])
         fgpu.image = 'katgpucbf'
+        fgpu.fake_katcp_server_cls = FakeFgpuDeviceServer
```

Comment:
Is there not an equivalent of this for xbgpu?

Reply:
There probably will be eventually, if it needs to expose sensors. The default is `FakeDeviceServer` which is good enough to handle requests for which there is no reply, such as `capture-start` / `capture-stop`.